addons/hw_drivers/iot_handlers/interfaces/SerialInterface.py

```diff
@@ -10,9 +11,11 @@ class SerialInterface(Interface):
     connection_type = 'serial'
 
     def get_devices(self):
-        serial_devices = {}
-        for port in serial.tools.list_ports.comports():
-            serial_devices[port.device] = {
-                'identifier': port.device
-            }
+        serial_ports = serial.tools.list_ports.comports()
+        serial_devices = {
+            port.device: {'identifier': port.device}
+            for port in serial_ports
+            if re.search(r'ttyAMA\d+', port.device) is None
```

Comment:
Maybe you can perform a better filtre than a re.search:
here all value returned by port, one with a real USB cable (/dev/ttyUSB0) and another from the system /dev/ttyAMA10:

**hit**: choice `port.subsystem` to filter it `use-serial` <-> `amba`

```
for port in serial.tools.list_ports.comports():
        print(port.device)
        print(port.apply_usb_info())
        print(port.description)
        print(port.device_path)
        print(port.hwid)
        print(port.interface)
        print(port.location)
        print(port.manufacturer)
        print(port.name)
        print(port.pid)
        print(port.product)
        print(port.serial_number)
        print(port.subsystem)
        print(port.usb_description())
        print(port.usb_device_path)
        print(port.usb_info())
        print(port.usb_interface_path)
        print(port.vid)
        print('-----------')

```

/dev/ttyUSB0
None
USB2.0-Ser!
/sys/devices/platform/axi/1000120000.pcie/1f00200000.usb/xhci-hcd.0/usb1/1-2/1-2:1.0/ttyUSB0
USB VID:PID=1A86:7523 LOCATION=1-2
None
1-2
None
ttyUSB0
29987
USB2.0-Ser!
None
usb-serial
USB2.0-Ser!
/sys/devices/platform/axi/1000120000.pcie/1f00200000.usb/xhci-hcd.0/usb1/1-2
USB VID:PID=1A86:7523 LOCATION=1-2
/sys/devices/platform/axi/1000120000.pcie/1f00200000.usb/xhci-hcd.0/usb1/1-2/1-2:1.0
6790
-----------
/dev/ttyAMA10
None
ttyAMA10
/sys/devices/platform/soc/107d001000.serial
USB VID:PID=0000:0000
None
None
None
ttyAMA10
None
None
None
amba
ttyAMA10
None
USB VID:PID=0000:0000
None
None
-----------